### ISSUE TITLE
client.sendFile // Not Working Anymore

### ISSUE THREAD

**Issue by LucasDibz** [smashah/sulla#94]
**Describe the bug**
client.sendFile stopped working with update, was working with version "sulla-hotfix": "1.6.05".
No error is thrown.
**Steps to Reproduce**

**create() code**
```javascript
fs.readFile(`${folder}${file}`, (err, data) => {
                    try {
                        client.sendFile(
                            'xxx@c.us',
                            `data:application/pdf;base64,${data.toString('base64')}`,
                            filename,
                            `Caption`
                        );
                    } catch (error) {
                        console.log(error);
                    };
                });
```
**DEBUG INFO**
This is the info printed to the console when you start your app. It should look like this
```
Debug Info {
  WA_VERSION: '0.4.612',
  PAGE_UA:
   'WhatsApp/0.4.607 Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
  SULLA_HOTFIX_VERSION: '1.7.5',
  BROWSER_VERSION: 'HeadlessChrome/79.0.3945.0' }
```

**Host (please complete the following information):**
 - OS: Ubuntu 18.04 LTS

**Comment by smashah** [comment]
Hi @LucasDibz 

Thanks for following the template. You have not shared your create() code. Can you do that, please?

Here's an example:
https://github.com/smashah/sulla/issues/96#issue-560091558

I have tried with PDFs on the latest version and it is working fine. sendFile code has not changed for a very long time.

**Comment by smashah** [comment]
Just FYI, i would suggest against using headless chrome

**Comment by LucasDibz** [comment]
That's the whole code:
`const startBot = (client) => {
                fs.readFile(`${folder}${file}`, (err, data) => {
                    try {
                        client.sendFile(
                            tel,
                            `data:application/pdf;base64,${data.toString('base64')}`,
                            filename,
                            `Caption`
                        );
                        client.sendText(tel, 'test'); //this works
                    } catch (error) {
                        console.error(error);
                    };
                });       
};
sulla.create().then(client => startBot(client));`

> Just FYI, i would suggest against using headless chrome

How could i do that? Kinda new to programming (~1year) and 4 months into NodeJs.

Thanks for your time.

**Comment by guijob** [comment]
Here neither `sendFile` or `sendImage` seems to work:

**create() code**
```
sulla.create('session', {
  executablePath: '/usr/bin/google-chrome',
  qrRefreshS: 60 * 60 * 24 * 7,
  headless: false,
}).then(client => {
```
I've tried without options too:
```
sulla.create().then(client => {
```

**DEBUG INFO**
```
Debug Info { WA_VERSION: '0.4.612',
  PAGE_UA:
   'WhatsApp/0.4.607 Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',
  SULLA_HOTFIX_VERSION: '1.7.5',
  BROWSER_VERSION: 'HeadlessChrome/80.0.3987.0' }
```
I'm running in **Ubuntu 19.10**

No errors thrown in chrome console or in my application.

**Comment by haroldorodsilva** [comment]
@LucasDibz did you have find a solution?

**Comment by LucasDibz** [comment]
@haroldorodsilva not yet, trying with diff create setting...still no success... maybe something about puppeteer?

**Comment by smashah** [comment]
@LucasDibz what puppeteer v are you running?

**Comment by LucasDibz** [comment]
@smashah sullas default,
 "_requested": {
    "type": "version",
    "registry": true,
    "raw": "puppeteer@2.0.0",
    "name": "puppeteer",
    "escapedName": "puppeteer",
    "rawSpec": "2.0.0",
    "saveSpec": null,
    "fetchSpec": "2.0.0"
  },

**Comment by smashah** [comment]
@LucasDibz can you run the demo, send it a pdf then see if it replies with the pdf?

**Comment by haroldorodsilva** [comment]
I am testing with the clone of the project and works fine. I used this code below:

  client.onMessage(async message => {
    try {
      const isConnected = await client.isConnected();
      console.log('TCL: start -> isConnected', isConnected);

      if (message.body.toUpperCase() === 'image'.toUpperCase()) {
        const image = await utils.getFileInBase64('image.png');

        console.log('send image to', message.from, image === '');

        await client.sendImage(
          message.from,
          // `data:image/png;base64,${image}`,
          String(image),
          'image.png',
          `Try to send image`
        );
      }
      //
    } catch (error) {
      console.log('TCL: start -> error', error);
    }
  });

I'll try using with npm and with the dist of the clone now.

**Comment by haroldorodsilva** [comment]
Now I had created a new project and make a clone of sulla inside this project. I used the script build to generate the foldes dist and used this folder.

const { create } = require("./sulla/dist");
const utils = require("./utils");

const startBot = client => {
  client.onMessage(async message => {
    try {
      const isConnected = await client.isConnected();
      console.log(" start -> isConnected", isConnected);

      if (message.body.toUpperCase() === "image".toUpperCase()) {
        const image = await utils.getFileInBase64("image.png");

        console.log("send image to", message.from, image === "");

        await client.sendImage(
          message.from,
          // `data:image/png;base64,${image}`,
          image,
          "image.png",
          `Try to send image`
        );
      }
      //
    } catch (error) {
      console.log("TCL: start -> error", error);
    }
  });
};
create().then(client => startBot(client));

With this way the it is not works.

**Comment by haroldorodsilva** [comment]
with package don't work too:

const { create } = require("sulla-hotfix");

**Comment by haroldorodsilva** [comment]
@smashah Can you try using download the library from npm and using files .js ?

because cloning the project and running ts-node on demo is working.

**Comment by cucabr** [comment]
the same problem occurs here. any ideas on how to solve?

**Comment by smashah** [comment]
I've moved totally to using ts-node for everything. I'd suggest migrating to that if it works until this issue gets sorted.

**Comment by cucabr** [comment]
@haroldorodsilva I'm a noob with ts-node. do you know how do i import the modules and be able to execute the code that you posted? I would be very grateful if you could help me.

**Comment by smashah** [comment]
@cucabr same as demo/index.ts

**Comment by cucabr** [comment]
around here, even using the ts-node with project clone (actual), it's not ok. nothing happens. and no errors appear either. sad. I discovered this repository today.

**Comment by haroldorodsilva** [comment]
The test that I think must be made is create a project with typescript and install the package with npm and try.

@cucabr não sou tão bom com typescript, procurei na Net apenas para testar. Tem que instalar o typescript e o ts-node e depois só rodar: ts-node demo/Index.js...

**Comment by haroldorodsilva** [comment]
> around here, even using the ts-node with project clone (actual), it's not ok. nothing happens. the image is not sent and no errors appear either. sad. I discovered this repository today.

I'll post here later the code latter

**Comment by haroldorodsilva** [comment]
> around here, even using the ts-node with project clone (actual), it's not ok. nothing happens. the image is not sent and no errors appear either. sad. I discovered this repository today.

I'll post here later the code latter

**Comment by haroldorodsilva** [comment]
> around here, even using the ts-node with project clone (actual), it's not ok. nothing happens. the image is not sent and no errors appear either. sad. I discovered this repository today.

You're right, I updated and did not send the image. @smashah 

I deleted the folder and cloned version 1.7.6 again and also did not send the image. I downloaded 1.7.5 in the releases and made the same change and it didn't work anymore.

Now I don't understand what happened. I will attach my files here from 1.7.6

**Comment by haroldorodsilva** [comment]
[sulla_176.zip](https://github.com/smashah/sulla/files/4176346/sulla_176.zip)

**Comment by smashah** [comment]
@LucasDibz @haroldorodsilva @cucabr try v1.7.7 it should be fixed now.

Let me know how it goes. Thanks.

**Comment by haroldorodsilva** [comment]
@smashah In version 1.7.7 it worked perfectly both in the demo of the clone, via npm and using the compiled dist.

Thanks

**Comment by LucasDibz** [comment]
@smashah Perfect!

thanks!

**Comment by gildastema** [comment]
Error: Evaluation failed: ReferenceError: chatId is not defined. ChatId not define.

**Comment by smashah** [comment]
@gildastema latest version? I'll fix that today. Thanks for pointing it out.

**Comment by haroldorodsilva** [comment]
@smashah the pull request #116 I have tested here with the alterations and works fine

**Comment by smashah** [comment]
@haroldorodsilva next v will include the fix in the above pull request

**Comment by smashah** [comment]
@haroldorodsilva @gildastema 

Fix is live in v1.7.12

thanks
